packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -505,6 +508,31 @@ contract L2Genesis is Deployer {
         _setImplementationCode(Predeploys.SUPERCHAIN_WETH);
     }
 
+    /// @notice This predeploy is following the safety invariant #1.
+    ///         This contract has no initializer.
+    function setOptimismSuperchainERC20Factory() internal {
+        _setImplementationCode(Predeploys.OPTIMISM_SUPERCHAIN_ERC20_FACTORY);
+    }
+
+    /// @notice This predeploy is following the safety invariant #2.
+    function setOptimismSuperchainERC20Beacon() internal {
+        address superchainERC20Impl = Predeploys.OPTIMISM_SUPERCHAIN_ERC20;
+        console.log("Setting %s implementation at: %s", "OptimismSuperchainERC20", superchainERC20Impl);
+        vm.etch(superchainERC20Impl, vm.getDeployedCode("OptimismSuperchainERC20.sol:OptimismSuperchainERC20"));
+
+        OptimismSuperchainERC20Beacon beacon = new OptimismSuperchainERC20Beacon(superchainERC20Impl);
+        console.log(
+            "Setting %s implementation at: %s",
+            "OptimismSuperchainERC20Beacon",
+            Predeploys.OPTIMISM_SUPERCHAIN_ERC20_BEACON
+        );
+        vm.etch(Predeploys.OPTIMISM_SUPERCHAIN_ERC20_BEACON, address(beacon).code);
```

Comment:
I believe that this line is what is causing the issue in CI.

The error:

```
predeploy 4200000000000000000000000000000000000027 is missing from L2 genesis allocs
```

Is from `op-chain-ops/genesis/layer_two.go:69`. I do not believe we want to `etch` the predeploy address - there should be a proxy there, we should set the implementation in the proxy to point to the beacon implementation. This will allow us to modify the implementation in the proxy address to a new address which will be a contract with a different `implementation()` beacon return value